vpd-manager/src/parser.cpp

```diff
@@ -71,6 +78,10 @@ int Parser::updateVpdKeyword(const types::WriteVpdParams& i_paramsToWriteData)
             "Error while updating keyword's value on hardware path " +
             m_vpdFilePath + ", error: " + std::string(l_exception.what()));
         // TODO : Log PEL
+
+        // Disable Reboot Guard.
+        dbusUtility::DisableRebootGuard();
```

Comment:
this disable guard is called several times in the same API, in call error scenario.

instead should we have one outer try & catch block for this API, and call this API once ?

Reply:
As per discussion : Have a outer `try` `catch` block and remove all `return -1` paths. Do everything in `catch`

Reply:
Done.